HHS/simpler-grants-gov
# [Issue #3043] Next login flow client

## Summary
Fixes #3043

### Time to review: __20 mins__

## Changes proposed
Adds a context provider and hook to allow client components to access up to date logged in user information. 

A NextJS route was added to return user data decrypted from the session cookie passed up from the client.

Some updates were made to the feature flags table in order for that page to act as a proof of concept for the functionality. These changes should be reverted before the feature goes live, but in the meantime are useful for testing.

Some feature flags and loading spinner code was moved to a more suitable location.

## Context for reviewers
### Test steps

1. run `npm run dev` on this branch in a new private window
3. visit http://localhost:3000/dev/feature-flags
4. _VERIFY_: you see a loading spinner followed by a page with a message saying "not logged in"
2. visit http://localhost:3000/api/auth/callback?token=anyStringYouWant2Use
3. visit http://localhost:3000/dev/feature-flags or hit "back" on your browser
4. _VERIFY_: feature flags page loads with message "Logged in with token: anyStringYouWant2Use"
5. refresh page with network tab of dev tools open
6. _VERIFY_: only one request to /api/auth/session is made
7. throw an error from line 8 of the session route handler file
8. refresh
9. _VERIFY_: the error details are printed out

## Additional information
### Follow up

* https://github.com/HHS/simpler-grants-gov/issues/3242
